### ISSUE TITLE
[BUG]: Deployment on Databricks

### ISSUE THREAD

**Issue by Yergat** [dotnet/spark#121]
Hi,

I followed the instructions provided on this [link](https://github.com/dotnet/spark/tree/master/deployment#databricks) to deploy my Spark .NET app on **azure databricks** but I can't get it to execute. I used Visual Studio 2019 to create my .NET Core app, I published using VS and zipped the content of the publish and copied the zip file to dbfs.

The first question I have, I setup the **_db-init.sh_** and **_install-worker.sh_** as explained and the cluster start without errors, is there any way to access the workers to verify if the installation was done successfully?

The second question I have, I created a job using the [Set Jar](https://github.com/dotnet/spark/tree/master/deployment#using-set-jar), what am I supposed to set the arguments **_<your-app-main-class>_**, I set it to **_MY_APP_NAME.dll_** which is in the zipped file. Is this correct?

This is the errors I'm getting with Spark 2.4.0:
ERROR Uncaught throwable from user code: org.apache.spark.SparkUserAppException: User application exited with 2

This is the errors I'm getting with Spark 2.4.2:
ERROR Uncaught throwable from user code: java.lang.IllegalArgumentException: Unsupported spark version used: 2.4.0. Normalized spark version used: 2.4.0. Supported versions: 2.3.0, 2.3.1, 2.3.2, 2.3.3

I tried on both _Databricks Runtime: 5.3 (Scala 2.11, Spark 2.4.0)_ and _Databricks Runtime: 5.4 Beta (Scala 2.11, Spark 2.4.2)_ but they both failed.

I also tried both microsoft-spark versions **_microsoft-spark-2.3.x-0.2.0.jar_** and **_microsoft-spark-2.4.x-0.2.0.jar_**, but again, no success.

Thanks in advance.

Mike

**Comment by imback82** [comment]
Can you please attach the full log for the run against Spark 2.4.0 using microsoft-spark-2.4.x-0.2.0.jar? (Please check [this](https://github.com/dotnet/spark/blob/master/docs/release-notes/0.2/release-0.2.md) for supported Spark versions). 

You may be hitting an issue fixed in this [PR](https://github.com/dotnet/spark/pull/92) and you can try to set this at the start of your application:
```C#
Environment.SetEnvironmentVariable("spark.master", "spark://");
```

**Comment by imback82** [comment]
> The second question I have, I created a job using the [Set Jar](https://github.com/dotnet/spark/tree/master/deployment#using-set-jar), what am I supposed to set the arguments ****, I set it to **_MY_APP_NAME.dll_** which is in the zipped file. Is this correct?

@suhsteve what is `<your-app-main-class>` here? That would be just **_MY_APP_NAME_** (the executable name right?)

**Comment by suhsteve** [comment]
> > The second question I have, I created a job using the [Set Jar](https://github.com/dotnet/spark/tree/master/deployment#using-set-jar), what am I supposed to set the arguments ****, I set it to **_MY_APP_NAME.dll_** which is in the zipped file. Is this correct?
> 
> @suhsteve what is `<your-app-main-class>` here? That would be just **_MY_APP_NAME_** (the executable name right?)

Yes, that will be the application binary contained in the .zip

**Comment by Yergat** [comment]
Thanks for the quick reply guys,

I attached the full log files here:
[stderr.txt](https://github.com/dotnet/spark/files/3212204/stderr.txt)
[stdout.txt](https://github.com/dotnet/spark/files/3212205/stdout.txt)
[log4j-active.txt](https://github.com/dotnet/spark/files/3212206/log4j-active.txt)

Here are the details of the cluster and jar files used:
Databricks Runtime version:
5.3 (includes Apache Spark 2.4.0, Scala 2.11)

Job:
Set Jar: microsoft_spark_2_4_x_0_2_0.jar
Main Class: org.apache.spark.deploy.DotnetRunner
Arguments: ["/dbfs/apps/ApacheSparkfordotNET.zip","ApacheSparkfordotNET.dll"]

ApacheSparkfordotNET.zip is uploaded to dbfs, I attached a screenshot of the files within the zip file, as seen here:
![ApacheSparkfordotNET zip content](https://user-images.githubusercontent.com/8135670/58254440-bdf37380-7d38-11e9-81fb-64be0a76221b.png)

I don't have an executable, the .NET Core app only generated dll's. I published the .NET Core app locally and my laptop using Visual Studio 2019 and I zipped the content before copying it to dbfs.

**Comment by imback82** [comment]
> I don't have an executable, the .NET Core app only generated dll's. I published the .NET Core app locally and my laptop using Visual Studio 2019 and I zipped the content before copying it to dbfs.

If you [published](https://github.com/dotnet/spark/tree/master/deployment#preparing-your-spark-net-app) correctly, you should see ApacheSparkfordotNet exectuable. Can you double check?

**Comment by Yergat** [comment]
> ApacheSparkfordotNet

You're right Tim, the VS publish was not giving the expected result, I had to run the dotnet publish command to get it to work.

I also tested the code with and without the following command, and you're right, I did have to include this in my code in order for it to run correctly. So when this fix is out, I won't be needing this code?

> Environment.SetEnvironmentVariable("spark.master", "spark://");
 
Thanks again for all your help.
